cmd/argocd/commands/project.go

```diff
@@ -590,38 +590,73 @@ func NewProjectRemoveSourceNamespace(clientOpts *argocdclient.ClientOptions) *co
 	return command
 }
 
-func modifyResourcesList(list *[]metav1.GroupKind, add bool, listDesc string, group string, kind string) bool {
-	if add {
-		for _, item := range *list {
-			if item.Group == group && item.Kind == kind {
-				fmt.Printf("Group '%s' and kind '%s' already present in %s resources\n", group, kind, listDesc)
-				return false
-			}
+func addResourceToAllowList(list []metav1.GroupKind, listDesc string, group string, kind string) ([]metav1.GroupKind, bool) {
+	for _, item := range list {
+		if item.Group == group && item.Kind == kind {
```

Comment:
Have you considered using `silces.Contains` https://pkg.go.dev/slices#Contains instead of iterating?
It's more elegant IMHO